### PR TITLE
docs: Add 'deposit and withdraw' guide to cookbook

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- This PR adds a TS SDK counterpart of the [deposit and withdraw guide in the Rust SDK cookbook](https://fuellabs.github.io/fuels-rs/v0.33.0/cookbook/deposit-and-withdraw.html).

- This PR is _related_ to #629 but does _not_ close it since we can still add more guides to the cookbook.

- ~~Marking this as a draft since the code in the guide right now is not being imported from any unit tests as suggested by Cameron in the original issue. Will push an update.~~

### CODE REVIEW

**Review by arboleya — CHANGES_REQUESTED** [review]

**Review by arboleya — APPROVED** [review]

File: packages/fuel-gauge/test-projects/liquidity-pool/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[project]
+authors = ["Dhaiwat Pandya"]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "liquidity-pool"
```
[the remark this answers is not in the document]
Comment (arboleya) [reply packages/fuel-gauge/test-projects/liquidity-pool/Forc.toml:5]: Good one. These can also be trimmed down, then.

https://github.com/FuelLabs/fuels-ts/blob/d0c74cd59f249ce1bdf113ce6f1744dcf48fc3e7/packages/fuel-gauge/test-projects/collision_in_fn_names/Forc.toml#L1-L5

https://github.com/FuelLabs/fuels-ts/blob/d0c74cd59f249ce1bdf113ce6f1744dcf48fc3e7/packages/fuel-gauge/test-projects/token_contract/Forc.toml#L1-L5

https://github.com/FuelLabs/fuels-ts/blob/d0c74cd59f249ce1bdf113ce6f1744dcf48fc3e7/packages/fuel-gauge/test-projects/token_abi/Forc.toml#L1-L5

File: packages/fuel-gauge/test-projects/token_contract/src/main.sw [hunk 2]
```diff
@@ -36,3 +37,4 @@ impl Token for Contract {
         msg_amount()
     }
 }
+// #endregion
```
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply packages/fuel-gauge/test-projects/token_contract/src/main.sw:40]: @camsjams not sure if I follow you here. The `token_contract` folder does have a [`Forc.toml`](https://github.com/FuelLabs/fuels-ts/blob/dp/issue-629/packages/fuel-gauge/test-projects/token_contract/Forc.toml)

**Review by arboleya — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
Excellent work @Dhaiwat10!

**Review by digorithm — APPROVED** [review]
:shipit:

**Pull request merged**